src/org/sosy_lab/java_smt/test/SanitizerTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * This file is part of JavaSMT,
+ * an API wrapper for a collection of SMT solvers:
+ * https://github.com/sosy-lab/java-smt
+ *
+ * SPDX-FileCopyrightText: 2024 Dirk Beyer <https://www.sosy-lab.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.sosy_lab.java_smt.test;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.sosy_lab.java_smt.api.BooleanFormula;
+import org.sosy_lab.java_smt.api.SolverException;
+
+public class SanitizerTest extends SolverBasedTest0.ParameterizedSolverBasedTest0 {
+  @Before
+  public void init() {
+    requireParser();
+    requireIntegers();
+  }
+
+  @Test
+  public void validLogicTest() throws SolverException, InterruptedException {
+    BooleanFormula expected = mgr.parse("(declare-const v Int)(assert (= v 3))");
+    BooleanFormula broken =
+        mgr.parse("(set-logic QF_ALL)" + "(declare-const v Int)" + "(assert (= v 3))");
+    assertThatFormula(expected).isEquivalentTo(broken);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
Please use `assertThrows(IllegalArgException.class, () -> mgr.parse(...))` if possible. This simplifies the test.

Same for following tests.